spec_02.md

```diff
@@ -0,0 +1,86 @@
+- Spec: 2
+- Title: Methods on figure objects
+- Version: 0.1
+- Last-Modified: 2019-07-19
+- Authors: [James Bednar](),
+           [David Hoese](),
+           [Jon Mease](jon.mease@gmail.com),
+           [Julia Signell](jsignell@gmail.com),
+- Status: Active
+- Type: Standards
+- Content-Type: text/markdown
+- Created: 2019-07-19
+
+## Abstract
+This spec describes a proposed shared API for accessing objects from various plotting libraries.
+
+In this notebook we will use **figure** to refer to any object that has a visible representation but which can also be saved, exported, etc. This includes generic visualizations or animation that may or may not be a "plot" (axes, ticks, etc)?
+
+When detailing a method, this spec uses FIGURE to refer to any specific figure object and LIBRARY to refer to any specific visualization library.
+
+## If a method doesn't make sense
+If a given operation doesn't make sense for that library, then it can satisfy the spec by simply having that method return a message to that effect ("JSON support not available in LIBRARY").
+
+## Jupyter methods
+Every library that supports rendering in Jupyter should support the various IPython rich display methods, i.e. `repr_html`, `repr_png`, `_ipython_display_`.
+
+## `.show()`
+Every figure should have the ability to render itself.
+
+### Proposal
+ 1) The top-level figure class has a `.show()` method that can be called without arguments to display the figure as a side-effect.
+ 2) The optional `renderer` kwarg can be used to override the current default renderer. e.g.:
+    - `FIGURE.show(renderer='png')` to display the figure as a static png image/
+    - `FIGURE.show(renderer='browser')` to display the figure in a browser tab. This works in non-jupyter/IPython contexts.
```

Comment:
What level of interactivity should be required from this `show` interface? And should there be a defined list of "preferred" rendering options that a library should support with defined names (ex. so one library doesn't use `jpg` while another uses `jpeg`).

Does `browser` mean interactive javascript based display? Or could it mean open a browser tab and display the PNG version of this figure? Does the browser rendering open a new tab if in a jupyter notebook environment or does it display it inline in the notebook cell's output?

Reply:
I would argue for a minimal standard for `show` that can be met by all libraries, and thus would not assume any interactivity or JavaScript.

Reply:
should this maybe be `output` instead of `renderer` so that it can match `.save()`?

Reply:
should the `output` option just be `'html'` then and a browser tab is implicitly where html would be displayed?

Reply:
I think that makes sense.